### PR TITLE
WIP: Add ampgen

### PR DESCRIPTION
**Pull request by dylanjaide**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/ampgen/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,339 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (chrisburr) [inline recipes/ampgen/LICENSE.txt:1]: You shouldn't need this file, conda-build will automatically look in the source tarball for matching files (you still need the licence_file field in the YAML)

File: recipes/ampgen/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - make
+    - root_base
+    - root-binaries
+    - root-dependencies
```
Comment (chrisburr) [inline recipes/ampgen/meta.yaml:24]: This should only be in `host`
```suggestion
```

File: recipes/ampgen/build.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+mkdir build
+cd build
+
+echo ${PREFIX}
+
+cmake \
+    -DCMAKE_INSTALL_PREFIX="${PREFIX}" \
+    -DCMAKE_INSTALL_RPATH="${PREFIX}/lib" \
+    -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=ON \
+    -DAMPGENROOT_CMAKE="${PREFIX}" \
```
Comment (chrisburr) [inline recipes/ampgen/build.sh:13]: This like is the problem, looks like it needs patching to something like `CMAKE_INSTALL_BINDIR`: https://github.com/GooFit/AmpGen/blob/d2c5bb112318ff0465debae9bb8dbff286a49a21/CMakeLists.txt#L167
```suggestion
```

**Comment by dylanjaide** [comment]
The current issue in the linux build is due to the value of AMPGENROOT not being updated after the work directory is moved at the start of the test stage:
```
2021-04-16T14:30:23.7181125Z TEST START: /home/conda/staged-recipes/build_artifacts/linux-64/ampgen-2.1-h635ac2c_0.tar.bz2
2021-04-16T14:30:24.2535251Z Renaming work directory '/home/conda/staged-recipes/build_artifacts/ampgen_1618582780625/work' to '/home/conda/staged-recipes/build_artifacts/ampgen_1618582780625/work_moved_ampgen-2.1-h635ac2c_0_linux-64'
2021-04-16T14:30:24.2537495Z INFO conda_build.utils:shutil_move_more_retrying(2077): Renaming work directory '/home/conda/staged-recipes/build_artifacts/ampgen_1618582780625/work' to '/home/conda/staged-recipes/build_artifacts/ampgen_1618582780625/work_moved_ampgen-2.1-h635ac2c_0_linux-64'
2021-04-16T14:30:24.2539252Z INFO conda_build.utils:shutil_move_more_retrying(2084): shutil.move(work)=/home/conda/staged-recipes/build_artifacts/ampgen_1618582780625/work, dest=/home/conda/staged-recipes/build_artifacts/ampgen_1618582780625/work_moved_ampgen-2.1-h635ac2c_0_linux-64)
2021-04-16T14:30:24.2540801Z shutil.move(work)=/home/conda/staged-recipes/build_artifacts/ampgen_1618582780625/work, dest=/home/conda/staged-recipes/build_artifacts/ampgen_1618582780625/work_moved_ampgen-2.1-h635ac2c_0_linux-64)
```
So the initial value of AMPGENROOT of `/home/conda/staged-recipes/build_artifacts/ampgen_1618582780625/work/build/bin` seems to be fine at compile time, but becomes incorrect for the tests. Is there a way to turn off the renaming of this directory, or a good way to update the var for the tests only?

The issue in the osx build seems to caused by a lot of package conflicts, but I've looked at some of them and it doesn't look like there should be any issues. For example:
```
Package libtiff conflicts for:
root-binaries -> afterimage[version='>=1.21,<1.22.0a0'] -> libtiff[version='>=4.0.9,<5.0a0']
root-binaries -> libtiff[version='>=4.0.10,<5.0a0|>=4.1.0,<5.0a0|>=4.2.0,<5.0a0']
```
I don't understand why these conflict - libtiff v4.0.10, for example, should satisfy both requirements if I understand the syntax here correctly?

**Review by carterbox — COMMENTED** [review]
@dylanjaide, conda packages need to be portable, so if your binaries rely on hard-coded paths, they will not work because the conda-environment will have a different path on every machine. If your binary needs to find an environment variable, you can add [activate and deactivate scripts](https://docs.conda.io/projects/conda-build/en/latest/resources/activate-scripts.html#) to your package. These scripts are run whenever a conda environment is activated that contains your package. For example, the conda-forge c compiler package will export the CC environment variable.

**Pull request closed without merging**